src/util/duration.h

```diff
@@ -136,21 +136,21 @@ class Duration : public DurationBase {
     // Returns a Duration object representing a duration of 'seconds'.
     template<typename T>
     static constexpr Duration fromSeconds(T seconds) {
-        return Duration(static_cast<qint64>(seconds * kNanosPerSecond));
+        return Duration(static_cast<double>(seconds * kNanosPerSecond));
```

Comment:
seconds should be casted to a double first.
Do we still need a template? I think If we make the seconds a double we can even remove the cast.

Reply:
kNanosPerSecond should be also a double to avoid loss of precision warnings.

Reply:
Done.